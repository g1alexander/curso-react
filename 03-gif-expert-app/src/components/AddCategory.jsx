import { useState } from "react";
import PropTypes from "prop-types";

export function AddCategory({ setCategories }) {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        value={inputValue}
        onChange={handleInputChange}
        type="text"
        name="add"
        id="add"
      />
    </form>
  );
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
