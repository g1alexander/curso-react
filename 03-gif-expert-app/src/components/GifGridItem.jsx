import PropTypes from "prop-types";

export function GifGridItem({ title, image }) {
  return (
    <div className="card animate__animated animate__bounce">
      <h3>{title}</h3>
      <img src={image} alt={title} />
    </div>
  );
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};
