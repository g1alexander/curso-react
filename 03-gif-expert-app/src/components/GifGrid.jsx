import PropTypes from "prop-types";
import { GifGridItem } from "./GifGridItem";
import { useFetchGif } from "../hooks/useFetchGif";

export function GifGrid({ category }) {
  const { data: images, loading } = useFetchGif(category);

  return (
    <>
      <h2>{category}</h2>

      {loading && (
        <p className="animate__animated animate__flash">cargando...</p>
      )}

      <section className="grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </section>
    </>
  );
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
