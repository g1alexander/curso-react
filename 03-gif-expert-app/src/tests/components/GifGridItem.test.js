import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("tests component GifGridItem", () => {
  const title = "un titulo",
    image = "http://localhost/hola.png";
  let wrapper = shallow(<GifGridItem title={title} image={image} />);

  test("debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe tener un h3 con el title", () => {
    const $h3 = wrapper.find("h3");

    expect($h3.text().trim()).toBe(title);
  });

  test("debe tener la imagen igual al url y alt de los props", () => {
    const $img = wrapper.find("img");

    //$img.props() == {src: "...", alt: "..."}

    expect($img.prop("src")).toBe(image);
    expect($img.prop("alt")).toBe(title);
  });

  test("debe tener el componente la clase animate__bounce", () => {
    const $div = wrapper.find("div");

    //alternativa: const className = $div.prop("className") == card animate__animated animate__bounce

    expect($div.hasClass("animate__bounce")).toBe(true);
  });
});
