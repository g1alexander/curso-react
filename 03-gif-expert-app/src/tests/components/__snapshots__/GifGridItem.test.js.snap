// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests component GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <h3>
    un titulo
  </h3>
  <img
    alt="un titulo"
    src="http://localhost/hola.png"
  />
</div>
`;
